core/src/main/java/org/elasticsearch/gateway/DanglingIndicesState.java

```diff
@@ -105,31 +105,31 @@ void findNewAndAddDanglingIndices(MetaData metaData) {
      * as dangled already.
      */
     Map<String, IndexMetaData> findNewDanglingIndices(MetaData metaData) {
-        final Set<String> indices;
+        final Set<String> indicesUUIDs;
         try {
-            indices = nodeEnv.findAllIndices();
+            indicesUUIDs = nodeEnv.findAllIndices();
```

Comment:
Now we load all the indices state files upfront and then add them to dangling indices if they are not present in the cluster state or are not identified as dangling already. This behaviour is different from before where we only used to load the state of indices that were relevant. We also don't try to rename the index according to the folder name, like before. Is this what you were getting at by "read some descriptor"?